Sources/Load.php

```diff
@@ -2469,18 +2469,20 @@ function loadTheme($id_theme = 0, $initialize = true)
 	);
 
 	// Add the JQuery library to the list of files to load.
-	if (isset($modSettings['jquery_source']) && $modSettings['jquery_source'] == 'cdn')
-		loadJavaScriptFile('https://ajax.googleapis.com/ajax/libs/jquery/' . JQUERY_VERSION . '/jquery.min.js', array('external' => true, 'seed' => false), 'smf_jquery');
+	$jQueryUrls = array ('google_cdn' => 'https://ajax.googleapis.com/ajax/libs/jquery/'. JQUERY_VERSION . '/jquery.min.js', 'jquery_cdn' => 'https://code.jquery.com/jquery-'. JQUERY_VERSION . '.min.js', 'microsoft_cdn' => 'https://ajax.aspnetcdn.com/ajax/jQuery/jquery-'. JQUERY_VERSION . '.min.js');
```

Comment:
> ... so if users have set the jQuery source as 'cdn' (Google) they will fallback to the local source until they resave their settings.

To avoid this, please change 'google_cdn' back to just 'cdn'.

If we were starting from scratch, I could see an argument for using jQuery's own CDN as the default rather than Google's, but at this point we should avoid changing the current behaviour unless necessary.

Reply:
The same will happen if they have the jquery source set to 'auto' or haven't even set the source by saving the page, they will now fallback to the local source instead of Google which was the previous fallback and I guess it would be advisable not to break that too?

Would adding the line `if (isset($modSettings['jquery_source']) && ($modSettings['jquery_source'] == 'cdn' || $modSettings['jquery_source'] == 'auto')) $modSettings['jquery_source'] = 'google_cdn';`or something along those lines, or simply expanding the array to include cdn and auto be a better idea to capture the old settings?

Reply:
In this case, the value `auto` means the admin said "I dunno, just use whatever the SMF devs have decided would be a good default", so that's not an issue. But the value `cdn` means that the admin specifically selected "Google CDN" as their jQuery source, so we should maintain that if we can.

... Also, the more I think about it, the more convinced I become that a CDN (I don't really care which one) should be the default, rather than the local file. Using a CDN for this is better both for the server and for the end user, so it should be the default unless the admin has a specific reason not to do so.